daffodil-core/src/main/scala/org/apache/daffodil/core/grammar/ElementBaseGrammarMixin.scala

```diff
@@ -81,6 +82,20 @@ trait ElementBaseGrammarMixin
     }
   }
 
+  final lazy val checkDelimitedLengthEVDP: Unit = {
+    if (
+      optionEmptyValueDelimiterPolicy.isDefined
+      && emptyValueDelimiterPolicy != EmptyValueDelimiterPolicy.Both
+      && (hasInitiator || hasTerminator)
```

Comment:
In general I think we want to avoid checking if a property is defined, since the property could be defined but not used, so we don't really care what its value is. EVDP is only used if there are initiator/terminator delimiters, so I think this could be something like:

```scala
if (
  (hasInitiator || hasterminator) &&
  emptyValueDelimiterPolicy != EmptyValueDelimiterPolicy.Both)
) {
  SDW(...)
}
```

This way we only care about the value of EVDP if there are delimiters, and if there are delimiters then it must exist and we warn if it's not "both".